### PR TITLE
 Implement edit comment action

### PR DESCRIPTION
**Pull request by ptomin**
Implement edit comment action (show declaration form for a while):
default

### CODE REVIEW

**Comment by ptomin** [comment]
I have created Comment Form (0d724bb) and used it instead of Declaration Form:
default

**Comment by ptomin** [comment]
I have created unit tests to verify `CommentFormInteractor` (5fe799b). Is it time to merge it, John?

**Comment by uxmal** [comment]
I'm reviewing now.

File: src/Gui/Windows/CombinedCodeViewInteractor.cs [hunk 1]
```diff
@@ -449,8 +451,21 @@ private void EditDeclaration()
...
+        private Address GetCommentAnchorAddress()
+        {
+            if (combinedCodeView.MixedCodeDataView.Focused)
+                return combinedCodeView.MixedCodeDataView.GetAnchorAddress();
+            return null;
+        }
+
         private void EditComment()
         {
+            var addr = GetCommentAnchorAddress();
```
Comment (uxmal) [inline src/Gui/Windows/CombinedCodeViewInteractor.cs]: `Edit comment` should be enabled for both the `MixedCodeControl` and the `NestedText` contrl on the right.

File: src/Gui/Windows/Controls/MixedCodeDataControl.cs [hunk 2]
```diff
@@ -52,9 +52,19 @@ public Program Program
             {
                 if (program != null && program.ImageMap != null)
                     program.ImageMap.MapChanged -= ImageMap_MapChanged;
+                if (program != null)
+                {
+                    program.User.Annotations
+                        .AnnotationChanged -= AnnotationChanged;
+                }
```
Comment (uxmal) [inline src/Gui/Windows/Controls/MixedCodeDataControl.cs]: Consider fusing the tests:
```
if (program != null) 
{
    if (program.ImageMap != null)
        program.ImageMap.MapChanged -= ImageMap_MapChanged;
    program.User.Annotations.AnnotationChanged -= AnnotationChanged;
}
```

File: src/Gui/Windows/Controls/MixedCodeDataControl.cs [hunk 3]
```diff
@@ -52,9 +52,19 @@ public Program Program
...
                 program = value;
                 if (program != null && program.ImageMap != null) 
                     program.ImageMap.MapChanged += ImageMap_MapChanged;
+                if (program != null)
```
Comment (uxmal) [inline src/Gui/Windows/Controls/MixedCodeDataControl.cs]: Same comment about fusing here.

File: src/Gui/Windows/Controls/MixedCodeDataControl.cs (hunk 2 above)
Comment (ptomin) [reply src/Gui/Windows/Controls/MixedCodeDataControl.cs]: Fixed. See 72c9976

File: src/Gui/Windows/Controls/MixedCodeDataControl.cs (hunk 3 above)
Comment (ptomin) [reply src/Gui/Windows/Controls/MixedCodeDataControl.cs]: See 72c9976

File: src/Gui/Windows/CombinedCodeViewInteractor.cs (hunk 1 above)
Comment (ptomin) [reply src/Gui/Windows/CombinedCodeViewInteractor.cs]: Do you mean `GetCommentAnchorAddress` should look like
```
private void EditComment()
{
    if (combinedCodeView.CodeView.Focused)
        return CodeView_GetAnchorAddress();
    if (combinedCodeView.MixedCodeDataView.Focused)
        return combinedCodeView.MixedCodeDataView.GetAnchorAddress();
    return null;
}
```

**Comment by uxmal** [comment]
Looks good.

File: src/Gui/Windows/CombinedCodeViewInteractor.cs (hunk 1 above)
Comment (uxmal) [reply src/Gui/Windows/CombinedCodeViewInteractor.cs]: Yes. There should be a way of editing the comments on the "right side" too

**Pull request merged**